python/rmm/_lib/device_buffer.pyx

```diff
@@ -77,25 +82,36 @@ cdef class DeviceBuffer:
         >>> import rmm
         >>> db = rmm.DeviceBuffer(size=5)
         """
-        cdef const void* c_ptr
+        cdef:
+            const void* c_ptr
+            cuda_stream_view c_stream_view
+            device_memory_resource* c_mr
+
+        if stream is None:
+            stream = DEFAULT_STREAM
+        if mr is None:
+            mr = get_current_device_resource()
+
+        # Save a reference to the MR and stream used for allocation
+        self.mr = mr
+        self.stream = stream
 
         with nogil:
             c_ptr = <const void*>ptr
+            c_stream_view = stream.view()
+            c_mr = mr.c_obj.get()
```

Comment:
Just a side note, it may be possible to use `get_mr` here, but that method would need to be marked `nogil`

https://github.com/rapidsai/rmm/blob/d94bdfd060c8c54379d01c21b8386492f36c9fd1/python/rmm/_lib/memory_resource.pyx#L167-L168

Reply:
Ah, that's probably what we should do